easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1102,11 +1133,20 @@ def _finalize_dependencies(self):
 
         for key in DEPENDENCY_PARAMETERS:
             # loop over a *copy* of dependency dicts (with resolved templates);
-            # to update the original dep dict, we need to index with idx into self._config[key][0]...
-            for idx, dep in enumerate(self[key]):
+            deps = self[key]
+
+            # to update the original dep dict, we need to get a reference with templating disabled...
+            deps_ref = self.get_ref(key)
+
+            # take into account that this *dependencies parameter may be iterated over
+            if key in self.iterate_options:
+                deps = flatten(deps)
+                deps_ref = flatten(deps_ref)
```

Comment:
@bartoldeman This looks wrong to me... `flatten` will create a new list, so `deps_ref` is most definitely no longer a reference to the original list...

Reply:
Actually, it's OK. `deps_ref` will be a new list, but the entries inside are still references to the original dependency `dict` values that we aim to modify, so it's all good...